# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/media-conduit/AudioConduit.cpp
# Commit: 25bf8e097e60
# Full Hash: 25bf8e097e604cca2842af6754c2c1698db5618a
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2019-10-29 21:27:48
# Description:
#   Bug 1543622 - Make number of channels out param of GetAudioFrame; r=pehrsons
#   
#   The number of channels is available in mAudioFrame in GetAudioFrame so
#   there is no reason to calculate it after the fact in MediaPipeline.
#   
# ==============================================================================

diff -r 4139bfe5e7f9 -r 25bf8e097e60 media/webrtc/signaling/src/media-conduit/AudioConduit.cpp
--- a/media/webrtc/signaling/src/media-conduit/AudioConduit.cpp	Tue Oct 29 16:04:29 2019 +0000
+++ b/media/webrtc/signaling/src/media-conduit/AudioConduit.cpp	Tue Oct 29 17:30:03 2019 +0000
@@ -583,7 +583,8 @@
 MediaConduitErrorCode WebrtcAudioConduit::GetAudioFrame(int16_t speechData[],
                                                         int32_t samplingFreqHz,
                                                         int32_t capture_delay,
-                                                        int& lengthSamples) {
+                                                        size_t& numChannels,
+                                                        size_t& lengthSamples) {
   CSFLogDebug(LOGTAG, "%s ", __FUNCTION__);
 
   // validate params
@@ -614,16 +615,23 @@
     return kMediaConduitSessionNotInited;
   }
 
-  int lengthSamplesAllowed = lengthSamples;
+  size_t lengthSamplesAllowed = lengthSamples;
   lengthSamples = 0;  // output paramter
 
   mRecvChannelProxy->GetAudioFrameWithInfo(samplingFreqHz, &mAudioFrame);
+  numChannels = mAudioFrame.num_channels_;
+
+  if (numChannels == 0) {
+    CSFLogError(LOGTAG, "%s Audio frame has zero channels", __FUNCTION__);
+    return kMediaConduitPlayoutError;
+  }
+
   // XXX Annoying, have to copy to our buffers -- refactor?
   lengthSamples = mAudioFrame.samples_per_channel_ * mAudioFrame.num_channels_;
   MOZ_RELEASE_ASSERT(lengthSamples <= lengthSamplesAllowed);
   PodCopy(speechData, mAudioFrame.data(), lengthSamples);
 
-  CSFLogDebug(LOGTAG, "%s GetAudioFrame:Got samples: length %d ", __FUNCTION__,
+  CSFLogDebug(LOGTAG, "%s GetAudioFrame:Got samples: length %zu ", __FUNCTION__,
               lengthSamples);
   return kMediaConduitNoError;
 }